templates/include/yarp/ast.h.erb

```diff
@@ -52,6 +52,24 @@ typedef enum {
 <%- end -%>
 } yp_node_type_t;
 
+// We store the node type enum in every node in the tree. We don't have nearly
+// as many node types as we do bits in an enum, so we're going to use the top
+// half of the enum to store flags about the node.
+static const unsigned int YP_NODE_FLAGS_SHIFT = sizeof(yp_node_type_t) * 8 / 2;
```

Comment:
Maybe bit fields could be used for this? Then you wouldn't need to mask manually

Reply:
This is causing compiler errors on linux-aarch64 like, as found by @andrykonchin:
```
In file included from include/yarp.h:5:0,
                 from src/unescape.c:1:
include/yarp/ast.h:349:50: error: conversion to ‘unsigned int’ from ‘int’ may change the sign of the result [-Werror=sign-conversion]
 static const unsigned int YP_NODE_FLAG_NEWLINE = 1 << YP_NODE_FLAGS_SHIFT;
                                                  ^
include/yarp/ast.h:349:50: error: initializer element is not constant
cc1: all warnings being treated as errors
```

Changing them to `#define` probably works around that.

But I feel it's really unsafe and most likely undefined behavior to set bits of an `enum` field like done in this PR.
And undefined behavior means the program can do random bad things, like segfaulting and assuming wrong things.
Could you try to use [C bit fields](https://www.tutorialspoint.com/cprogramming/c_bit_fields.htm) instead?